### ISSUE TITLE
Connection Failed: Please check your API key.

### ISSUE THREAD

**Issue by hakhan201** [vultr/whmcs-vultr#107]
Hello,

When I try to save my API key on my WHMCS site, I get an error saying "Connection Failed: Please check your API key.".

I disabled my server firewalls, and allowed all ip's to connect to the API from my vultr profile, however it did not help.

What could possibly be the issue?

I even regenerated the API key and tried to use it but the issue continued to remain.

Please help, I'm using WHMCS 8.1.3.

Regards,
Haider Ali

**Comment by Vsnumimmy** [comment]
Yes it was showing Invalid API Key and it was not generating the Config Options in Module section

The API key was correct

How to solve this issue

Latest WHMCS version 8.1.3

**Comment by sqpp** [comment]
Obviously you add the IP address of the WHMCS server in Vultr...

**Comment by Vsnumimmy** [comment]
> Obviously you add the IP address of the WHMCS server in Vultr...

Had tried whitelisting the Server IP/32 on Vultr API Even though it was not working and showing same error

**Comment by hakhan201** [comment]
> > Obviously you add the IP address of the WHMCS server in Vultr...
> 
> Had tried whitelisting the Server IP/32 on Vultr API Even though it was not working and showing same error

Same here :(

**Comment by sqpp** [comment]
Try to make a Curl Request to the Vultr API and see how does it responds, I believe something is blocking your connection from the WHMCS side.

**Comment by Vsnumimmy** [comment]
> Try to make a Curl Request to the Vultr API and see how does it responds, I believe something is blocking your connection from the server where WHMCS is hosted on.

Thanks I had solved it

I had made an CURL Request from server and it was connecting using IPV6 But the IPV6 was not whitelisted in the Vultr API.I  had enabled the IPV6 Address and it was working

**Comment by sqpp** [comment]
Glad to hear it man!

**Comment by amit510** [comment]
> > Try to make a Curl Request to the Vultr API and see how does it responds, I believe something is blocking your connection from the server where WHMCS is hosted on.
> 
> Thanks I had solved it
> 
> I had made an CURL Request from server and it was connecting using IPV6 But the IPV6 was not whitelisted in the Vultr API.I had enabled the IPV6 Address and it was working

Thank you for saving me troubleshooting time!
